---
title: "coseq_analysis"
author: "Agneesh Barua"
date: "4/4/2022"
output: html_document
---

```{r setup}
library(tidyverse)
library(coseq)
library(ggsci)
```

## Read in data
```{r,message=F}
final_orth_list<-read_csv("../data/orthologs/final_orth_list.csv")%>% drop_na() %>% pivot_longer(!Orthogroup)

#use dge gene, filter data to isolate genes that are consistently upregulated and downregulated in all samples. These are individual transcripts, not orthologs.
dge_up<-read_csv("../data/data_tables/dge_all_hosts_vs_non_hosts_sig.csv") %>%  dplyr::select(!fdr) %>% spread(sp,logFC) %>% drop_na() %>% filter(if_all(where(is.numeric), ~ .x > 0) | !if_any(where(is.numeric), ~ .x < 0))

dge_down<-read_csv("../data/data_tables/dge_all_hosts_vs_non_hosts_sig.csv") %>% dplyr::select(!fdr) %>% spread(sp,logFC) %>% drop_na() %>% filter(if_all(where(is.numeric), ~ .x < 0) | !if_any(where(is.numeric), ~ .x > 0))

#Obtain the orthogroups of the up and down regulated genes
dge_orths_up<-final_orth_list %>% filter(value %in% dge_up$gene) %>% distinct(Orthogroup)
dge_orths_down<-final_orth_list %>% filter(value %in% dge_down$gene) %>% distinct(Orthogroup)
```

## Input file for coseq

```{r,eval=F}
coseq_input<-dge_orths_up

dat<-read_csv("../data/orthologs/all_sp_all_tissue_exp_matrix_counts.csv") %>% filter(Orthogroup %in% coseq_input$Orthogroup) %>%  column_to_rownames("Orthogroup")

#get expression data for anemonefish hosting giant sea anemone
dat %>% dplyr::select(Enqua_tentacles1,
               Enqua_tentacles2,
               Enqua_tentacles3,
               Enqua_tentacles4,
               Enqua_tentacles5,
               Enqua_tentacles6,
               Mado_tentacle_i1,
               Mado_tentacle_i2,
               Mado_tentacle_o1,
               Mado_tentacle_o2,
               Stgag_tentacles1,
               Stgag_tentacles2,
               Stgag_tentacles3,
               Stgag_tentacles4,
               Stmer_tentacles1,
               Stmer_tentacles2,
               Stmer_tentacles3,
               Stmer_tentacles4,
               Sthad_tentacles1,
               Sthad_tentacles2,
               Sthad_tentacles3,
               Heau_tentacles1,
               Heau_tentacles2,
               Heau_tentacles3,
               Heau_tentacles4,
               Heau_tentacles5,
               Hecri_tentacles1,
               Hecri_tentacles2,
               Hecri_tentacles3,
               Hecri_tentacles4,
               Hemag_tentacles1,
               Hemag_tentacles2,
               Hemag_tentacles3,
               Hemag_tentacles4,
               Hemag_tentacles5) %>% round()->t

t[is.na(t)]<-0 #make NAs to to zero

#make these objects for coseq input
annot.df<-AnnotatedDataFrame(t(t) %>% as.data.frame()) 
exp.dat<-ExpressionSet(assayData = as.matrix(t))
counts <- exprs(exp.dat)
```

## Run analysis
```{r,eval=F}
runArcsin <- coseq(counts, K=2:10, model="Normal", transformation="arcsin",norm = "TMM", 
                   meanFilterCutoff=50, iter=100000,seed=0001)
runArcsin %>% saveRDS("../data/rds/runArcsin_dge_up.rds")
```

## Read results
```{r}
runArcsin<-readRDS("../data/rds/runArcsin_dge_up.rds")
#For downregulated data read in the runArcsin_dge_down.rds

#summary
summary(runArcsin) 
```

# Plots
```{r}
plot(runArcsin, graphs="boxplots",cond = sampleNames(exp.dat) %>% gsub(pattern = "_tentacles[0-9]",replacement = "") %>% gsub(pattern = "_tentacle_i[0-9]",replacement = "") %>% gsub(pattern = "_tentacle_o[0-9]",replacement = ""),add_lines = F)->p

p$boxplots+theme_minimal()+
    scale_fill_nejm()+
    theme(axis.ticks.x = element_blank(),
          axis.text.x = element_blank(),
          panel.grid = element_blank())

plot(runArcsin, graphs="profiles")->pp

## Covariance matrix for cluster 1
rho <- NormMixParam(runArcsin)$rho
rho1 <- rho[,,1]
colnames(rho1) <- rownames(rho1) <- paste0(colnames(rho1), "\n")

rho <- NormMixParam(runArcsin)$rho
rho2 <- rho[,,2]
colnames(rho1) <- rownames(rho1) <- paste0(colnames(rho1), "\n")

corrplot::corrplot(rho1,method = "color",tl.col = "black",tl.cex = 0.6,col = colorRampPalette(c("grey20", "white", "darkgreen"))(100))

corrplot::corrplot(rho2,method = "color",tl.col = "black",tl.cex = 0.6,col = colorRampPalette(c("grey20", "white", "darkgreen"))(100))
```

## Check if cor patterns in due to low number of genes or actually a biological trend captured by coseq
```{r}
#use the matrix from above chunk to summarise correlations
#use number of gene in cluster (46 and 176); The patterns in the coseq cluster are different from same number of random genes
corrplot::corrplot(cor(log10(t[c(floor(runif(46,1,222))),]+1)),
                   method = "color",tl.col = "black",tl.cex = 0.6,
                   col = colorRampPalette(c("grey20", "white", "darkgreen"))(100))
```

## Getting the clusters
```{r}
probapost <- assay(runArcsin)
c1<-probapost %>% as.data.frame() %>% rownames_to_column("OG") %>% filter(Cluster_1 >= 0.8)
c2<-probapost %>% as.data.frame() %>% rownames_to_column("OG") %>% filter(Cluster_2 >= 0.8)
```

# GO term enrichment
```{r,message=F}
library(GOstats)
library(GO.db)
library(GSEABase)
library(AnnotationDbi)
library(RSQLite)

annot<-read_csv("../data/data_tables/GO_annot_universe.csv")
annot<-annot %>% filter(ARGOT_PPV >= 0.5)
annot$goid<-sub("^","GO:",annot$goid)
annot<-annot %>% mutate(evidence = "IEA") %>% dplyr::select(go_id = goid, evidence,gene = qpid) #keep this order
goFrame<-GOFrame(as.data.frame(annot, organism = "hosts"))
goAllFrame<-GOAllFrame(goFrame)
gsc<-GeneSetCollection(goAllFrame,setType = GOCollection())
universe<-annot$gene
```

```{r,message=F,eval=F}
read_csv("../data/orthologs/final_orth_list.csv")%>% drop_na() -> final_orth_list
#get upshift gene

c1_id<-tibble(ids= (final_orth_list %>% filter(Orthogroup %in% c1$OG) %>% 
  dplyr::select(c(Orthogroup,enqua,heaur,hecri,hema,hetdac,macdac,stga,stha,stmer)) %>% 
  pivot_longer(cols = c(!Orthogroup)))$value)

up<-GSEAGOHyperGParams(name = "GO enrich", 
                             geneSetCollection = gsc,
                             geneIds = c1_id$ids,
                             universeGeneIds = universe,
                             ontology = "BP",
                             pvalueCutoff = 0.05,
                             conditional = F,
                             testDirection = "over")

Over_up<- hyperGTest(up)

Over_up.pv<-pvalues(Over_up)
pv.fdr<-p.adjust(Over_up.pv,"fdr")
summary(Over_up) %>% as_tibble() %>% mutate(FDR = pv.fdr[1:nrow(summary(Over_up) %>% as_tibble())]) %>% dplyr::select(GOBPID,FDR,Term,Size) %>% filter(FDR<=0.05) -> GOenrich
GOenrich %>% write_csv("./Transcriptomes/Annotations/GOenrich_coseq_upreg_cluster1.csv")
```
